test/collect.cc

```diff
@@ -37,8 +82,14 @@ TEST(Collect, CommonSetPass) {
   ASSERT_EQ(2, result.size());
   EXPECT_EQ(5, *result.begin());
   EXPECT_EQ(12, *++result.begin());
+
+  // The initial set should remain unchanged.
+  ASSERT_EQ(2, v.size());
+  EXPECT_EQ(5, *v.begin());
+  EXPECT_EQ(12, *++v.begin());
```

Comment:
Let's use an `EXPECT_THAT(v,  ElementsAre(5, 12))` or it's variants here and below please!